### PR TITLE
Allowing updating DataFile records via the API, e.g. to allow

### PR DESCRIPTION
**Pull request by wettenhj**
supplying a bogus MD5 sum initially to save time, and then
calculating the MD5 sum at the same time as doing the file upload.

### CODE REVIEW

**Comment by grischa** [comment]
@wettenhj while it doesn't matter much in this case, I would appreciate if you rebased your code instead of merging develop back in as a matter of habit, to avoid mess like #576 
No need to change it here, but in the future, please do. Happy to show you how to use `git rebase` if you're unfamiliar.

File: tardis/tardis_portal/api.py [hunk 1]
```diff
@@ -415,7 +415,14 @@ def update_detail(self, object_list, bundle):  # noqa # too complex
         elif isinstance(bundle.obj, DatasetParameter):
             return False
         elif isinstance(bundle.obj, DataFile):
-            return False
+            dataset = DatasetResource.get_via_uri(DatasetResource(),
+                                                  bundle.data['dataset'],
+                                                  bundle.request)
+            return all([
+                bundle.request.user.has_perm('tardis_portal.change_dataset'),
```
Comment (grischa) [inline tardis/tardis_portal/api.py]: datafile permissions automatically "bubble up", a single check should be enough, so just:
`bundle.request.user.has_perm('tardis_portal.change_datafile')`

**Comment by wettenhj** [comment]
Hi @grischa, happy to rebase rather than merge in future.  In this case, I was trying out the relatively new "Update branch" button on GitHub, described here: https://github.com/blog/2051-protected-branches-and-required-status-checks

**Comment by grischa** [comment]
hm, interesting. Looks like a useful feature... Maybe it works fine when all other history is quite “clean”, i.e. only your own commits

**Comment by grischa** [comment]
Thanks

**Pull request merged**